lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -358,27 +358,40 @@ pub trait SegmentOptimizer {
             PayloadFieldSchema,
         >::new()));
 
-        let proxies = optimizing_segments.iter().map(|sg| {
-            ProxySegment::new(
+        let mut proxies = Vec::new();
+        for sg in optimizing_segments.iter() {
+            let mut proxy = ProxySegment::new(
                 sg.clone(),
                 tmp_segment.clone(),
                 proxy_deleted_points.clone(),
                 proxy_created_indexes.clone(),
                 proxy_deleted_indexes.clone(),
-            )
-        });
+            );
+            // Wrapped segment is fresh, so it has no operations
+            // Operation with number 0 will be applied
+            let op_num = 0;
+            proxy.replicate_field_indexes(op_num)?;
```

Comment:
if we have a second call of `replicate_field_indexes`, we don't need this first one. On second `replicate_field_indexes` all indices will be created

Reply:
pls. read the comment. 

Second call will require the whole storage to be locked. We want to minimize the this possibility by creating the existing indexes in advance.

Reply:
Oh, there is explanation in comment